pkg/hive/hive_test.go

```diff
@@ -0,0 +1,199 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package hive
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"testing"
+
+	ma "github.com/multiformats/go-multiaddr"
+
+	pb "github.com/ethersphere/bee/pkg/hive/pb"
+	"github.com/ethersphere/bee/pkg/logging"
+	"github.com/ethersphere/bee/pkg/p2p"
+	"github.com/ethersphere/bee/pkg/p2p/protobuf"
+	"github.com/ethersphere/bee/pkg/p2p/streamtest"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+func TestInit(t *testing.T) {
+	logger := logging.New(ioutil.Discard, 0)
+	connectionManager := &ConnectionManagerMock{}
+	peerSuggester := &PeerSuggesterMock{}
+	addressFinder := &AddressFinderMock{}
+
+	// this is the receiving side
+	nodeReceiver := New(Options{
+		PeerSuggester: peerSuggester,
+		AddressFinder: addressFinder,
+		Logger:        logger,
+	})
+
+	// setup the stream recorder to record stream data
+	streamer := streamtest.New(
+		streamtest.WithProtocols(nodeReceiver.Protocol()),
+	)
+
+	nodeInit := New(Options{
+		Streamer:          streamer,
+		ConnectionManager: connectionManager,
+		Logger:            logger,
+	})
+
+	t.Run("OK", func(t *testing.T) {
+		suggesterPeers := make(map[int][]p2p.Peer)
+		addr1, addr2, addr3 :=
+			swarm.MustParseHexAddress("ca1e9f3938cc1425c6061b96ad9eb93e134dfe8734ad490164ef20af9d1cf59a"),
+			swarm.MustParseHexAddress("ca1e9f3938cc1425c6061b96ad9eb93e134dfe8734ad490164ef20af9d1cf59b"),
+			swarm.MustParseHexAddress("ca1e9f3938cc1425c6061b96ad9eb93e134dfe8734ad490164ef20af9d1cf59c")
+
+		suggesterPeers[0] = []p2p.Peer{
+			{Address: addr1},
+			{Address: addr2},
+		}
+		suggesterPeers[1] = []p2p.Peer{
+			{Address: addr3},
+		}
+
+		peerSuggester.Peers = suggesterPeers
+		addresses := make(map[string]ma.Multiaddr)
+		addr1Multi, err := ma.NewMultiaddr("/ip4/1.1.1.1")
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		addr2Multi, err := ma.NewMultiaddr("/ip4/1.1.1.1")
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		addr3Multi, err := ma.NewMultiaddr("/ip4/1.1.1.1")
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		addresses[addr1.String()] = addr1Multi
+		addresses[addr2.String()] = addr2Multi
+		addresses[addr3.String()] = addr3Multi
```

Comment:
Isn't the addresses map holding only one key as all three addresses are the same?

Reply:
Ah, a typo, when I was switching to multiaddr, wanted to make them different 🙈